Package a local directory code_source_path for AI Runtime tasks (#6110)
databricks/cli: .nextchanges/bundles/ai-runtime-code-source-dir.md

> An `ai_runtime_task.code_source_path` that points at a local directory is now packaged into a tarball during `bundle deploy` (honoring `.gitignore` and `sync.include`/`sync.exclude`, content-addressed so unchanged code skips re-upload), placed in the bundle and uploaded by bundle file sync, and rewritten to the uploaded workspace path. Pip dependencies are taken from the job's serverless `environments` spec. Complements the existing pre-built-tarball path by letting users point at a source directory directly.

Reviewer: ```suggestion
For jobs where `ai_runtime_task.code_source_path` is a relative path to a local directory, the directory is now packaged into a tarball (honoring `.gitignore` and `sync.include`/`sync.exclude`), uploaded during deployment, and `code_source_path` is rewritten to the uploaded workspace path.
```

overly verbose

> An `ai_runtime_task.code_source_path` that points at a local directory is now packaged into a tarball during `bundle deploy` (honoring `.gitignore` and `sync.include`/`sync.exclude`, content-addressed so unchanged code skips re-upload), uploaded to the user's workspace code snapshot directory, and rewritten to the uploaded path. A `requirements.yaml` derived from the job's serverless `environments` is written next to the task's `command_path`. Complements the existing pre-built-tarball path by letting users point at a source directory directly.

Reviewer: requirements.yaml is no longer synthesised
